packages/next-swc/crates/core/src/next_ssg.rs

```diff
@@ -132,12 +132,34 @@ impl Fold for Analyzer<'_> {
 
     fn fold_export_named_specifier(&mut self, s: ExportNamedSpecifier) -> ExportNamedSpecifier {
         if let ModuleExportName::Ident(id) = &s.orig {
-            self.add_ref(id.to_id());
+            let ssg_exports = &["getStaticProps", "getStaticPaths", "getServerSideProps"];
+
+            if !ssg_exports.contains(&&*id.sym) {
+                self.add_ref(id.to_id());
+            }
         }
 
         s
     }
 
+    fn fold_export_decl(&mut self, s: ExportDecl) -> ExportDecl {
+        if let Decl::Var(d) = &s.decl {
+            if d.decls.is_empty() {
+                return s;
+            }
+
+            if let Pat::Ident(id) = &d.decls[0].name {
+                let ssg_exports = &["getStaticProps", "getStaticPaths", "getServerSideProps"];
```

Comment:
Should we define this at the top-level so it can be shared instead of re-defining it each time or could we re-use the `is_data_identifier` method here?

Reply:
Probably yes. I wrote PR to gut check if this sounds correct idea first before trying to refactor.